tests/lib/tools/fs-state

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: fs-state wait-for-file [filename] [iterations] [sleep-time]"
+    echo "usage: fs-state mock-dir [dirname]"
+    echo "usage: fs-state restore-dir [dirname]"
+    echo "usage: fs-state mock-file [filename]"
+    echo "usage: fs-state restore-file [filename]"
+}
+
+wait_for_file() {
```

Comment:
I left a comment below that I think 'wait_for_file` should be reduced to just `retry test -e ...` in actual test code, so that `fs-state` does not have a think wrapper around another helper. I would remove this function entirely.